tests/unit/type/stream-cgroups.tcl

```diff
@@ -205,11 +205,34 @@ start_server {
         $rd close
     }
 
+    test {Blocking XREADGROUP for stream that ran dry (issue #5299)} {
+        set rd [redis_deferring_client]
+
+        # Add a entry then delete it, now stream's last_id is 666.
+        r DEL mystream
+        r XGROUP CREATE mystream mygroup $ MKSTREAM
+        r XADD mystream 666 key value
+        r XDEL mystream 666
+
+        # Serve it synchronously if the ID specified was not the special `>` ID.
+        foreach id {0 600 666 700} {
+            $rd XREADGROUP GROUP mygroup myconsumer BLOCK 10 STREAMS mystream $id
+            assert_equal [$rd read] {{mystream {}}}
```

Comment:
IIUC the original complaint was:
1. it returns right away instead of blocking
2. it returns the a stream with an empty list instead of nil.

let's test the fist complaint too.
i.e. set it to block for a long time using a deferring client, see that it got blocked, and then push something in (xadd) to release it)

Reply:
Yes, actually `$rd XREADGROUP GROUP mygroup myconsumer BLOCK 10 STREAMS mystream $id`
it will return right away, i see the BLOCK 10 may cause misunderstanding, i changed it to BLOCK 0

About the release part, i think i missed this one, it would be better to add, thanks